redis/src/connection.rs

```diff
@@ -810,17 +817,34 @@ impl Connection {
         // messages are received until the _subscription count_ in the responses reach zero.
         let mut received_unsub = false;
         let mut received_punsub = false;
-        loop {
-            let res: (Vec<u8>, (), isize) = from_redis_value(&self.recv_response()?)?;
+        if self.resp3 {
```

Comment:
Why was this resolved?

Reply:
repeated logic was moved to `is_pub_sub_state_cleared`. Since the async logic requires using async functions, the whole block couldn't be implemented in the same way.